src/main/java/org/jenkinsci/plugins/workflow/steps/CoreStep.java

```diff
@@ -48,42 +49,44 @@
 /**
  * A step that runs a {@link SimpleBuildStep} as defined in Jenkins core.
  */
-public final class CoreStep extends AbstractStepImpl {
+public final class CoreStep extends Step {
 
     public final SimpleBuildStep delegate;
 
     @DataBoundConstructor public CoreStep(SimpleBuildStep delegate) {
         this.delegate = delegate;
     }
 
-    private static final class Execution extends AbstractSynchronousNonBlockingStepExecution<Void> {
+    @Override public StepExecution start(StepContext context) throws Exception {
+        return new Execution(delegate, context);
+    }
+
+    private static final class Execution extends SynchronousNonBlockingStepExecution<Void> {
 
-        @Inject private transient CoreStep step;
-        @StepContextParameter private transient Run<?,?> run;
-        @StepContextParameter private transient FilePath workspace;
-        @StepContextParameter private transient Launcher launcher;
-        @StepContextParameter private transient TaskListener listener;
+        private transient final SimpleBuildStep delegate;
```

Comment:
Have the option of simplifying `StepExecution` state, especially in cases where the `Step` had only one field.